### ISSUE TITLE
Version 1.0

### ISSUE THREAD

**Issue by cgt** [cgt/go-mwclient#7]
I am working on a version of go-mwclient in branch `next` which I intend to tag as v1.0.0 once it is merged into `master`.

The following changes are currently planned for v1, some are already implemented:

* Support for OAuth authentication with the "owner-only consumer" method as described here: https://www.mediawiki.org/wiki/OAuth/Owner-only_consumers

* Default to formatversion=2 of the JSON API described here: https://www.mediawiki.org/wiki/API:JSON_version_2. One particularly nice improvement in version 2 of the JSON API is that lists of pages are represented as JSON arrays rather than an object where the keys are page IDs. This makes it much easier to decode into a struct. It shall be possible to override this by setting formatversion=1 manually in your params.Values parameter objects.

* `New()` does not return an error if the `userAgent` parameter is empty.

* `Login` and `Logout` will no longer set an API assertion level (used to assert that the client is logged in or has bot rights). This must now be done manually.

* `Logout` may be changed to actually handle errors rather than just firing off the logout request blindly. The method was only implemented for completeness, I don't recall ever actually using it.

* Verify that CAPTCHA error handling actually works (or get rid of it).

Some of these are breaking changes. Once v1 is tagged, go-mwclient shall follow semantic versioning as described on semver.org, with the exception that breaking changes may be introduced without a major version bump if changes to the MediaWiki API itself require it. go-mwclient shall be kept up to date and compatible with MediaWiki.

I'm not sure if anyone else but me is still using package, but I figured I would give people a chance to object to breaking changes or suggest other changes before v1. I don't have a plan for when v1 will be released, but it will likely be in the near future, possibly before the end of 2016.

**Comment by garyhouston** [comment]
Thanks for continuing to maintain this library. It's still used by my bot, at least, which has reached about 340k edits. I don't have any objection to breaking changes, assuming the new API works as well as the old. When authentication stopped working a few months ago I just switched to a bot password.
